graftm/tree_decorator.py

```diff
@@ -1,309 +1,187 @@
+        __init__ of this function
         
-        Inputs: 
-            taxonomy_list: array
-                - List of split taxonomy strings to be tansposed
-        Outputs: 
-            transposed_taxonomy: array
-                - As per above description
-        '''
-        transposed_taxonomy = []
-        for index in range(0, 7):
+        Parameters
+        ----------
 
-            rank_taxonomy_list = [taxonomy_entry[index] 
-                                    for taxonomy_entry in taxonomy_list]
-            transposed_taxonomy.append(set(rank_taxonomy_list))
-        return transposed_taxonomy
-    
-    def _write_tree(self, output):
-        '''Just Writes tree to file uisng scikit-bio'''
-        logging.info("Writing decorated tree to file: %s" % (output))
-        self.tree.write(
-                    output,
-                    format = "newick"   
-                        )
         
-    def _write_consensus_strings(self, output):
-        '''Writes the taxonomy of each leaf to a file'''
-        logging.info("Writing decorated taxonomy to file: %s" % (output))
-        with open(output, 'w') as out:
-            for tip in self.tree.tips():
-                ancestors = tip.ancestors()
-                
-                ancestor_tax = []
-                for ancestor in ancestors:
-                    if ancestor.name:
-                        split_node_name = ancestor.name.split(':')
-                        if len(split_node_name) == 2:
-                            ancestor_tax.append(split_node_name[1])
-                        elif len(split_node_name) == 1:
-                            pass
-                        else:
-                            raise Exception("Malformed node name: %s" % ancestor.name)
-                        
-                tax_list = list(reversed(ancestor_tax))
-                tax_name = tip.name.replace(" ", "_")
-                if len(tax_list) < 1:
-                    logging.debug("Species %s not decorated within the tree, using provided tax string as substitute" % (tax_name))
-                    if tax_name in self.taxonomy:
-                        tax_string = '; '.join(self.taxonomy[tax_name])
-                    else:
-                        logging.warning("No taxonomy found for species %s!" % (tax_name))
-                        tax_string = "Unknown"
-                else:
-                    tax_string = '; '.join(tax_list)
-                output_line = "%s\t%s\n" % (tax_name, tax_string)
-                out.write(output_line)
-    
-    def _get_tax_index(self, ancestry):
+        taxonomy: TaxonomyObject
+                 
+        Returns
+        -------
+        skbio TreeNode object
         '''
-        Iterate through ancestor nodes to find the current taxonomy index (i.e.
-        the rank that should be assigned to the current node.
+        self.max_taxonomy_index=taxonomy.max
+        consistency_check = 1
+        seen_taxonomy_rank = {}
```

Comment:
I make a habit of naming hashes to be key_to_value, so in this case maybe `taxonomy_to_next_unique`, maybe you want to follow this, maybe not